api/hypershift/v1beta1/nodepool_types.go

```diff
@@ -879,15 +879,17 @@ type AzureNodePoolPlatform struct {
 	// +required
 	VMSize string `json:"vmsize"`
 
-	// ImageID is the id of the image to boot from. If unset, the default image at the location below will be used and
-	// is expected to exist: subscription/<subscriptionID>/resourceGroups/<resourceGroupName>/providers/Microsoft.Compute/images/rhcos.x86_64.vhd.
-	// The <subscriptionID> and the <resourceGroupName> are expected to be the same resource group documented in the
-	// Hosted Cluster specification respectively, hcluster.Spec.Platform.Azure.SubscriptionID and
-	// hcluster.Spec.Platform.Azure.ResourceGroupName.
+	// ImageID is the id of the image to boot from. If both ImageID and MarketplaceImageInfo are provided, ImageID will
+	// take precedence according to CAPZ - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/f49b5ac4fe91ac2d2439f202f9b7ce5bae17470e/azure/converters/image.go#L29.
```

Comment:
is there any value in having both specified? can't we otherwise have them in their own struct wit a union discriminator?

Reply:
We would want ImageID for self-managed azure I would think.

Reply:
you also need the go markers see https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/1027-api-unions/README.md#design-details
See also https://github.com/search?q=repo%3Aopenshift%2Fapi%20unionDiscriminator&type=code for multiple examples